pkg/storage/common/common.go

```diff
@@ -529,7 +531,101 @@ func IsBlobReferenced(imgStore storageTypes.ImageStore, repo string,
 		return false, err
 	}
 
-	return isBlobReferencedInImageIndex(imgStore, repo, digest, index, log)
+	return IsBlobReferencedInImageIndex(imgStore, repo, digest, index, log)
+}
+
+/* Garbage Collection */
+
+func AddImageManifestBlobsToReferences(imgStore storageTypes.ImageStore,
+	repo string, mdigest godigest.Digest, refBlobs map[string]bool, log zlog.Logger,
+) error {
+	manifestContent, err := GetImageManifest(imgStore, repo, mdigest, log)
+	if err != nil {
+		log.Error().Err(err).Str("repository", repo).Str("digest", mdigest.String()).
+			Msg("gc: failed to read manifest image")
+
+		return err
+	}
+
+	refBlobs[mdigest.String()] = true
+	refBlobs[manifestContent.Config.Digest.String()] = true
+
+	// if there is a Subject, it may not exist yet and that is ok
```

Comment:
I do have a concern with regards to this.

Let's say we release an image, we tag it with `latest`, and sign it.
At some point we want to push a newer image and retag that one as `latest`.
Since the older image has a digest referenced in the signature, it will never be GCed, is that correct? Do we want that? I don't think so, because we'd never GC anything as they will be referenced forever? 

My understanding is it is good to GC images if they are only referenced as subject by artifacts, and are not tagged, and they are not referenced in an index.
We should also be ok with GCing artifacts which have missing subjects after the GC delay interval passes.

Reply:
We need an image retention policy, separate from GC for cleaning up images without tags.